metrics/open-tracing-backend/src/main/scala/sttp/client/opentracing/OpenTracingBackend.scala

```diff
@@ -0,0 +1,74 @@
+package sttp.client.opentracing
+
+import io.opentracing.tag.Tags
+import io.opentracing.Tracer
+import io.opentracing.propagation.Format
+import sttp.client.monad.MonadError
+import sttp.client.ws.WebSocketResponse
+import sttp.client.{FollowRedirectsBackend, NothingT, Request, Response, SttpBackend}
+import sttp.client.monad.syntax._
+import scala.collection.JavaConverters._
+
+class OpenTracingBackend[F[_], S] private (delegate: SttpBackend[F, S, NothingT], tracer: Tracer)
+    extends SttpBackend[F, S, NothingT] {
+
+  private implicit val _monad: MonadError[F] = responseMonad
+
+  override def send[T](request: Request[T, S]): F[Response[T]] =
+    responseMonad
+      .eval {
+        tracer
+          .buildSpan(
+            request
+              .tag(OpenTracingBackend.OperationIdRequestTag)
+              .getOrElse("default-operation-id")
```

Comment:
maybe this can be configurable? and a function - each time returning a different (by default randomised) id?

Reply:
According to this it should not be randomized IMO: 
https://github.com/yurishkuro/opentracing-tutorial/tree/master/java/src/main/java/lesson01#annotate-the-trace-with-tags-and-logs

Reply:
ah ok :)